main.py

```diff
@@ -1544,6 +1591,41 @@ def test_category(self):
             getattr(self, f"check_pair_idx_{category}")((psi[a], psi[b]), idx)
 
 
+class Test_Integral_Driven_Categories(Test_Minimal):
+    def setUp_categories(self):  # sort integral categories for minimal test case
+        psi, d_two_e_integral = self.psi_int
+        minimal_integral_categories = defaultdict(list)
+        for idx4 in d_two_e_integral:
+            idx = compound_idx4_reverse(idx4)
+            cat = integral_category(*idx)
+            minimal_integral_categories[cat].append(idx)
+        return minimal_integral_categories
+
+    def get_det_driven_integrals(self, category):
```

Comment:
Do you mind doing the logic as `setUp_categories`?
Aka returning a `dict {category: list(ab,idx,phase)}`?

Reply:
Done!